### ISSUE TITLE
(Windows/MSVC) Static library not generated on first build, ok on second.

### ISSUE THREAD

**Issue by Klaim** [build2/build2#71]
Context
---------

This issue appears with build2 `v0.12.0` on Windows only (with last stable msvc, didn't try other compilers). It does not appear on Linux with gcc9 (with the same build2 version).
```
$ b --version
build2 0.12.0
libbutl 0.12.0
host x86_64-microsoft-win32-msvc14.2
```

I didn't try previous build2 versions with this specific code, but I suspect that the issue is older than `v0.12.0`.

How To Reproduce
---------------------

On Windows (I used git-bash):
```
git clone https://github.com/klaim/lua-build2.git
cd lua-build2
git checkout 4e7e7c95dc791a07c99b9a92ac9f4324db5a0cff
bdep init -C ../build-msvc cc config.install.root=../install
b install    # 1
b install    #  2
```

The problem occurs on the install steps.
You need to `b uninstall clean` before retrying these steps or the ones listed in the analysis below.

Observed
-----------

The first `b install` triggers an error `/bin/install: cannot stat '..\build-msvc\lua\lua.lib': No such file or directory` because the file does not exist. Although we assume that it should have been built.

The second `b install` solves the problem by actually generating the file.

```
klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ b install    # 1
mkdir ..\build-msvc\lua\fsdir{src\}
c src\c{lzio}@..\build-msvc\lua\src\
c src\c{lapi}@..\build-msvc\lua\src\
c src\c{lauxlib}@..\build-msvc\lua\src\
c src\c{lbitlib}@..\build-msvc\lua\src\
c src\c{luac}@..\build-msvc\lua\src\
c src\c{lua}@..\build-msvc\lua\src\
c src\c{lbaselib}@..\build-msvc\lua\src\
c src\c{lzio}@..\build-msvc\lua\src\
c src\c{lvm}@..\build-msvc\lua\src\
c src\c{lcode}@..\build-msvc\lua\src\
c src\c{lcorolib}@..\build-msvc\lua\src\
c src\c{lvm}@..\build-msvc\lua\src\
c src\c{lctype}@..\build-msvc\lua\src\
c src\c{ldblib}@..\build-msvc\lua\src\
c src\c{ldebug}@..\build-msvc\lua\src\
c src\c{ldo}@..\build-msvc\lua\src\  
c src\c{ldump}@..\build-msvc\lua\src\
c src\c{lfunc}@..\build-msvc\lua\src\
c src\c{lgc}@..\build-msvc\lua\src\
c src\c{lutf8lib}@..\build-msvc\lua\src\
c src\c{linit}@..\build-msvc\lua\src\
c src\c{lutf8lib}@..\build-msvc\lua\src\
c src\c{liolib}@..\build-msvc\lua\src\
c src\c{llex}@..\build-msvc\lua\src\
c src\c{lmathlib}@..\build-msvc\lua\src\
c src\c{lundump}@..\build-msvc\lua\src\
c src\c{lmem}@..\build-msvc\lua\src\ 
c src\c{loadlib}@..\build-msvc\lua\src\
c src\c{lundump}@..\build-msvc\lua\src\
c src\c{lobject}@..\build-msvc\lua\src\
c src\c{lopcodes}@..\build-msvc\lua\src\
c src\c{loslib}@..\build-msvc\lua\src\
c src\c{lparser}@..\build-msvc\lua\src\
c src\c{ltm}@..\build-msvc\lua\src\
c src\c{lstate}@..\build-msvc\lua\src\
c src\c{ltm}@..\build-msvc\lua\src\  
c src\c{ltablib}@..\build-msvc\lua\src\
c src\c{lstring}@..\build-msvc\lua\src\
c src\c{lstrlib}@..\build-msvc\lua\src\
c src\c{ltable}@..\build-msvc\lua\src\
c src\c{ltablib}@..\build-msvc\lua\src\
c src\c{lapi}@..\build-msvc\lua\src\ 
c src\c{lauxlib}@..\build-msvc\lua\src\
c src\c{lbaselib}@..\build-msvc\lua\src\
c src\c{lbitlib}@..\build-msvc\lua\src\
c src\c{ltable}@..\build-msvc\lua\src\
c src\c{lcode}@..\build-msvc\lua\src\
c src\c{lcorolib}@..\build-msvc\lua\src\
c src\c{lctype}@..\build-msvc\lua\src\
c src\c{ldblib}@..\build-msvc\lua\src\
c src\c{ldebug}@..\build-msvc\lua\src\
c src\c{ldo}@..\build-msvc\lua\src\  
c src\c{ldump}@..\build-msvc\lua\src\
c src\c{lfunc}@..\build-msvc\lua\src\
c src\c{lstrlib}@..\build-msvc\lua\src\
c src\c{lgc}@..\build-msvc\lua\src\  
c src\c{linit}@..\build-msvc\lua\src\
c src\c{liolib}@..\build-msvc\lua\src\
c src\c{llex}@..\build-msvc\lua\src\ 
c src\c{lmathlib}@..\build-msvc\lua\src\
c src\c{lmem}@..\build-msvc\lua\src\ 
c src\c{loadlib}@..\build-msvc\lua\src\
c src\c{lstring}@..\build-msvc\lua\src\
c src\c{lobject}@..\build-msvc\lua\src\
c src\c{lopcodes}@..\build-msvc\lua\src\
c src\c{loslib}@..\build-msvc\lua\src\
c src\c{lparser}@..\build-msvc\lua\src\
c src\c{lstate}@..\build-msvc\lua\src\
ar ..\build-msvc\lua\liba{lua}       
ld ..\build-msvc\lua\exe{luac}       
ld ..\build-msvc\lua\libs{lua}       
ln ..\build-msvc\lua\exe{luac} -> .\
ld ..\build-msvc\lua\exe{lua}
ln ..\build-msvc\lua\exe{lua} -> .\
install ..\install\
install ..\install\share\
install ..\install\share\doc\
install ..\install\share\doc\lua\
install manifest{manifest}@..\build-msvc\lua\
install ..\install\include\
install src\h{lauxlib}@..\build-msvc\lua\src\
install src\h{lua}@..\build-msvc\lua\src\
install src\h{luaconf}@..\build-msvc\lua\src\
install src\h{lualib}@..\build-msvc\lua\src\
install src\hxx{lua}@..\build-msvc\lua\src\
install ..\build-msvc\lua\liba{lua}
/bin/install: cannot stat '..\build-msvc\lua\lua.lib': No such file or directory
install ..\install\bin\
install ..\build-msvc\lua\libs{lua}
install ..\build-msvc\lua\exe{lua}
install doc\doc{contents.html}@..\build-msvc\lua\doc\
install doc\doc{index.css}@..\build-msvc\lua\doc\
install doc\doc{logo.gif}@..\build-msvc\lua\doc\
install doc\doc{lua.1}@..\build-msvc\lua\doc\
install doc\doc{lua.css}@..\build-msvc\lua\doc\
install doc\doc{luac.1}@..\build-msvc\lua\doc\
install doc\doc{manual.css}@..\build-msvc\lua\doc\
install doc\doc{manual.html}@..\build-msvc\lua\doc\
install doc\doc{osi-certified-72x60.png}@..\build-msvc\lua\doc\
install doc\doc{readme.html}@..\build-msvc\lua\doc\
install doc{README}@..\build-msvc\lua\
info: failed to install ..\build-msvc\dir{lua\}

klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ b install    #  2
ar ..\build-msvc\lua\liba{lua}       
ld ..\build-msvc\lua\exe{luac}       
ln ..\build-msvc\lua\exe{luac} -> .\
install manifest{manifest}@..\build-msvc\lua\
install src\h{lauxlib}@..\build-msvc\lua\src\
install src\h{lua}@..\build-msvc\lua\src\
install src\h{luaconf}@..\build-msvc\lua\src\
install src\h{lualib}@..\build-msvc\lua\src\
install src\hxx{lua}@..\build-msvc\lua\src\
install ..\build-msvc\lua\liba{lua}
install ..\build-msvc\lua\libs{lua}
install ..\build-msvc\lua\exe{lua}
install ..\build-msvc\lua\exe{luac}
install doc\doc{contents.html}@..\build-msvc\lua\doc\
install doc\doc{index.css}@..\build-msvc\lua\doc\
install doc\doc{logo.gif}@..\build-msvc\lua\doc\
install doc\doc{lua.1}@..\build-msvc\lua\doc\
install doc\doc{lua.css}@..\build-msvc\lua\doc\
install doc\doc{luac.1}@..\build-msvc\lua\doc\
install doc\doc{manual.css}@..\build-msvc\lua\doc\
install doc\doc{manual.html}@..\build-msvc\lua\doc\
install doc\doc{osi-certified-72x60.png}@..\build-msvc\lua\doc\
install doc\doc{readme.html}@..\build-msvc\lua\doc\
install doc{README}@..\build-msvc\lua\
```

Expected
----------

 - The same result whatever the number of `b install`;
 - If the library's buildfile is correct, a systematic success, otherwise a systematic error (same as on linux/gcc);

Notes & Analysis
-------------------

 - The main symptom is that the `ar liba{lua}` step in the first `b install` does not seem to produce an output for some reason: 
```
$ ls ../build-msvc/lua/
build  liblua.shared.pc  liblua.static.pc  lua.dll  lua.dll.d  lua.dll.exp  lua.dll.lib  lua.exe  lua.exe.d  lua.exe.manifest  lua.lib.d  luac.exe  luac.exe.d  luac.exe.manifest  src  tests
```
then after the second `b install`:
```
 ls ../build-msvc/lua/
build  liblua.shared.pc  liblua.static.pc  lua.dll  lua.dll.d  lua.dll.exp  lua.dll.lib  lua.exe  lua.exe.d  lua.exe.manifest  lua.lib  lua.lib.d  luac.exe  luac.exe.d  luac.exe.manifest  src  tests
```
 - Successive `b update` show the same issue.
```
klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ b update
mkdir ..\build-msvc\lua\fsdir{src\}
mkdir ..\build-msvc\lua\tests\fsdir{basics\}
c src\c{ldump}@..\build-msvc\lua\src\
c src\c{lua}@..\build-msvc\lua\src\
c src\c{lzio}@..\build-msvc\lua\src\
c src\c{luac}@..\build-msvc\lua\src\
c src\c{lfunc}@..\build-msvc\lua\src\
c src\c{lbaselib}@..\build-msvc\lua\src\
c src\c{ldo}@..\build-msvc\lua\src\
c src\c{lgc}@..\build-msvc\lua\src\
c src\c{linit}@..\build-msvc\lua\src\
c src\c{lvm}@..\build-msvc\lua\src\
c src\c{liolib}@..\build-msvc\lua\src\
c src\c{llex}@..\build-msvc\lua\src\
c src\c{lmathlib}@..\build-msvc\lua\src\
c src\c{lbitlib}@..\build-msvc\lua\src\
c src\c{lmem}@..\build-msvc\lua\src\
c src\c{ldblib}@..\build-msvc\lua\src\
c src\c{loadlib}@..\build-msvc\lua\src\
c src\c{lutf8lib}@..\build-msvc\lua\src\
c src\c{lobject}@..\build-msvc\lua\src\
c src\c{lopcodes}@..\build-msvc\lua\src\
c src\c{loslib}@..\build-msvc\lua\src\
c src\c{lcode}@..\build-msvc\lua\src\
c src\c{lparser}@..\build-msvc\lua\src\
c src\c{lbitlib}@..\build-msvc\lua\src\
c src\c{lstate}@..\build-msvc\lua\src\
c src\c{lundump}@..\build-msvc\lua\src\
c src\c{lstring}@..\build-msvc\lua\src\
c src\c{lcorolib}@..\build-msvc\lua\src\
c src\c{lstrlib}@..\build-msvc\lua\src\
c src\c{ltable}@..\build-msvc\lua\src\
c src\c{ltablib}@..\build-msvc\lua\src\
c src\c{ltm}@..\build-msvc\lua\src\
c src\c{lbaselib}@..\build-msvc\lua\src\
c src\c{lctype}@..\build-msvc\lua\src\
c src\c{ldblib}@..\build-msvc\lua\src\
c src\c{ldebug}@..\build-msvc\lua\src\
c src\c{ldo}@..\build-msvc\lua\src\
c src\c{ldump}@..\build-msvc\lua\src\
c src\c{lfunc}@..\build-msvc\lua\src\
c src\c{lgc}@..\build-msvc\lua\src\
c src\c{linit}@..\build-msvc\lua\src\
c src\c{liolib}@..\build-msvc\lua\src\
c src\c{llex}@..\build-msvc\lua\src\
c src\c{lmathlib}@..\build-msvc\lua\src\
c src\c{lmem}@..\build-msvc\lua\src\
c src\c{loadlib}@..\build-msvc\lua\src\
c src\c{lobject}@..\build-msvc\lua\src\
c src\c{lopcodes}@..\build-msvc\lua\src\
c src\c{loslib}@..\build-msvc\lua\src\
c src\c{lparser}@..\build-msvc\lua\src\
c src\c{lstate}@..\build-msvc\lua\src\
c src\c{lstring}@..\build-msvc\lua\src\
c src\c{lstrlib}@..\build-msvc\lua\src\
c src\c{ltable}@..\build-msvc\lua\src\
c src\c{ltablib}@..\build-msvc\lua\src\
c src\c{ltm}@..\build-msvc\lua\src\
c src\c{lundump}@..\build-msvc\lua\src\
c src\c{lutf8lib}@..\build-msvc\lua\src\
c src\c{lvm}@..\build-msvc\lua\src\
c src\c{lapi}@..\build-msvc\lua\src\
c src\c{ldebug}@..\build-msvc\lua\src\
c src\c{lctype}@..\build-msvc\lua\src\
c src\c{lauxlib}@..\build-msvc\lua\src\
c src\c{lauxlib}@..\build-msvc\lua\src\
c src\c{lcode}@..\build-msvc\lua\src\
c src\c{lzio}@..\build-msvc\lua\src\
c++ tests\basics\cxx{driver}@..\build-msvc\lua\tests\basics\
c src\c{lapi}@..\build-msvc\lua\src\
c src\c{lcorolib}@..\build-msvc\lua\src\
ld ..\build-msvc\lua\libs{lua}
ar ..\build-msvc\lua\liba{lua}
ld ..\build-msvc\lua\exe{luac}
ld ..\build-msvc\lua\exe{lua}
ld ..\build-msvc\lua\tests\basics\exe{driver}
ln ..\build-msvc\lua\exe{luac} -> .\
ln ..\build-msvc\lua\tests\basics\exe{driver} -> tests\basics\
ln ..\build-msvc\lua\exe{lua} -> .\

klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ ls ../build-msvc/lua/
build  lua.dll  lua.dll.d  lua.dll.exp  lua.dll.lib  lua.exe  lua.exe.d  lua.exe.dlls  lua.exe.manifest  lua.lib.d  luac.exe  luac.exe.d  luac.exe.manifest  src  tests

klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ b update
ar ..\build-msvc\lua\liba{lua}
ld ..\build-msvc\lua\exe{luac}
ln ..\build-msvc\lua\exe{luac} -> .\

klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ ls ../build-msvc/lua/
build  lua.dll  lua.dll.d  lua.dll.exp  lua.dll.lib  lua.exe  lua.exe.d  lua.exe.dlls  lua.exe.manifest  lua.lib  lua.lib.d  luac.exe  luac.exe.d  luac.exe.manifest  src  tests

```
 - At first it looks like a target requirement tracking issue, though I can't spot errors in the buildfile, feel free to point me if there are;
 - It does not seem to be a concurrency/synchronization/race issue as using the serial mode will result in the same behavior:

```
klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ b install -s
mkdir ..\build-msvc\lua\fsdir{src\}
c src\c{lapi}@..\build-msvc\lua\src\
c src\c{lauxlib}@..\build-msvc\lua\src\
c src\c{lbaselib}@..\build-msvc\lua\src\
c src\c{lbitlib}@..\build-msvc\lua\src\ 
c src\c{lcode}@..\build-msvc\lua\src\  
c src\c{lcorolib}@..\build-msvc\lua\src\
c src\c{lctype}@..\build-msvc\lua\src\  
c src\c{ldblib}@..\build-msvc\lua\src\
c src\c{ldebug}@..\build-msvc\lua\src\
c src\c{ldo}@..\build-msvc\lua\src\   
c src\c{ldump}@..\build-msvc\lua\src\
c src\c{lfunc}@..\build-msvc\lua\src\
c src\c{lgc}@..\build-msvc\lua\src\  
c src\c{linit}@..\build-msvc\lua\src\
c src\c{liolib}@..\build-msvc\lua\src\
c src\c{llex}@..\build-msvc\lua\src\ 
c src\c{lmathlib}@..\build-msvc\lua\src\
c src\c{lmem}@..\build-msvc\lua\src\ 
c src\c{loadlib}@..\build-msvc\lua\src\
c src\c{lobject}@..\build-msvc\lua\src\
c src\c{lopcodes}@..\build-msvc\lua\src\
c src\c{loslib}@..\build-msvc\lua\src\
c src\c{lparser}@..\build-msvc\lua\src\
c src\c{lstate}@..\build-msvc\lua\src\
c src\c{lstring}@..\build-msvc\lua\src\
c src\c{lstrlib}@..\build-msvc\lua\src\
c src\c{ltable}@..\build-msvc\lua\src\
c src\c{ltablib}@..\build-msvc\lua\src\
c src\c{ltm}@..\build-msvc\lua\src\  
c src\c{lundump}@..\build-msvc\lua\src\
c src\c{lutf8lib}@..\build-msvc\lua\src\
c src\c{lvm}@..\build-msvc\lua\src\  
c src\c{lzio}@..\build-msvc\lua\src\ 
ar ..\build-msvc\lua\liba{lua}       
c src\c{lapi}@..\build-msvc\lua\src\ 
c src\c{lauxlib}@..\build-msvc\lua\src\
c src\c{lbaselib}@..\build-msvc\lua\src\
c src\c{lbitlib}@..\build-msvc\lua\src\
c src\c{lcode}@..\build-msvc\lua\src\
c src\c{lcorolib}@..\build-msvc\lua\src\
c src\c{lctype}@..\build-msvc\lua\src\
c src\c{ldblib}@..\build-msvc\lua\src\
c src\c{ldebug}@..\build-msvc\lua\src\
c src\c{ldo}@..\build-msvc\lua\src\  
c src\c{ldump}@..\build-msvc\lua\src\
c src\c{lfunc}@..\build-msvc\lua\src\
c src\c{lgc}@..\build-msvc\lua\src\  
c src\c{linit}@..\build-msvc\lua\src\
c src\c{liolib}@..\build-msvc\lua\src\
c src\c{llex}@..\build-msvc\lua\src\ 
c src\c{lmathlib}@..\build-msvc\lua\src\
c src\c{lmem}@..\build-msvc\lua\src\ 
c src\c{loadlib}@..\build-msvc\lua\src\
c src\c{lobject}@..\build-msvc\lua\src\
c src\c{lopcodes}@..\build-msvc\lua\src\
c src\c{loslib}@..\build-msvc\lua\src\
c src\c{lparser}@..\build-msvc\lua\src\
c src\c{lstate}@..\build-msvc\lua\src\
c src\c{lstring}@..\build-msvc\lua\src\
c src\c{lstrlib}@..\build-msvc\lua\src\
c src\c{ltable}@..\build-msvc\lua\src\
c src\c{ltablib}@..\build-msvc\lua\src\
c src\c{ltm}@..\build-msvc\lua\src\  
c src\c{lundump}@..\build-msvc\lua\src\
c src\c{lutf8lib}@..\build-msvc\lua\src\
c src\c{lvm}@..\build-msvc\lua\src\  
c src\c{lzio}@..\build-msvc\lua\src\ 
ld ..\build-msvc\lua\libs{lua}       
c src\c{lua}@..\build-msvc\lua\src\  
ld ..\build-msvc\lua\exe{lua}        
ln ..\build-msvc\lua\exe{lua} -> .\
c src\c{luac}@..\build-msvc\lua\src\
ld ..\build-msvc\lua\exe{luac}       
LINK : fatal error LNK1181: cannot open input file '..\build-msvc\lua\lua.lib'
info: failed to update (for install) ..\build-msvc\dir{lua\}

klaim@KLAIMDEV MINGW64 /e/Projects/build2-libs/lua/repro-install_from_git/lua-build2 ((4e7e7c9...))
$ b install -s
ar ..\build-msvc\lua\liba{lua}       
ld ..\build-msvc\lua\exe{luac}       
ln ..\build-msvc\lua\exe{luac} -> .\
install ..\install\
install ..\install\share\
install ..\install\share\doc\
install ..\install\share\doc\lua\
install manifest{manifest}@..\build-msvc\lua\
install ..\install\include\
install src\h{lauxlib}@..\build-msvc\lua\src\
install src\h{lua}@..\build-msvc\lua\src\
install src\h{luaconf}@..\build-msvc\lua\src\
install src\h{lualib}@..\build-msvc\lua\src\
install src\hxx{lua}@..\build-msvc\lua\src\
install ..\build-msvc\lua\liba{lua}
install ..\install\bin\
install ..\build-msvc\lua\libs{lua}
install ..\build-msvc\lua\exe{lua}
install ..\build-msvc\lua\exe{luac}
install doc\doc{contents.html}@..\build-msvc\lua\doc\
install doc\doc{index.css}@..\build-msvc\lua\doc\
install doc\doc{logo.gif}@..\build-msvc\lua\doc\
install doc\doc{lua.1}@..\build-msvc\lua\doc\
install doc\doc{lua.css}@..\build-msvc\lua\doc\
install doc\doc{luac.1}@..\build-msvc\lua\doc\
install doc\doc{manual.css}@..\build-msvc\lua\doc\
install doc\doc{manual.html}@..\build-msvc\lua\doc\
install doc\doc{osi-certified-72x60.png}@..\build-msvc\lua\doc\
install doc\doc{readme.html}@..\build-msvc\lua\doc\
install doc{README}@..\build-msvc\lua\
```
The linking error in the first install step of that last log can be reproduced by doing `b update -s`.

Here is the output of `b update -V -s`: https://pastebin.com/2cFw7Uq6

It looks like it could be due to an issue in `"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\bin\Hostx64\x64\lib.exe"`. If this is the source of the issue, maybe build2 can detect that the output was not generated before going farther? It might be expensive I guess.

 - I suspect that my recent change in the buildfiles of that library to compile it in C instead of C++ made the issue visible. Maybe it's related to the flags associated to C?

 - Side note but maybe relevant: there have always been weird behaviors from this package buildfile and I have been working on simplifying the it to reduce human error. Currently the main buildfile seems simple but some lines marked as TODO show weird behaviors that might (or not) be related to this problem. I planned to discuss these points before making issues, but this isssue seems to be clear enough to be reported.

**Comment by karen-arutyunov** [comment]
Thanks for the report.

The build2 bugfix version is staged.

**Comment by Klaim** [comment]
I confirm that the issue is fixed in `0.13.0-a.0.8fbc673998ed`
